dts: arm: st: h7: fix flash on M4 board targets

tomi-font opened the pull request:
The flash controller is nowadays supported on the M4 core.
Add the bank2-flash-size property to the board definitions as required by the STM32 H7 flash driver.

benjaminbjornsson commented:
Do you have an example of a test where the m4 is built and fails due to this?

I see the same problem running the following:

`west build -b arduino_portenta_h7/stm32h747xx/m4 -s zephyr/samples/drivers/flash_shell/`

I may be wrong here, but I don't think disabling the flash driver on board level is what we want.

An alternate solution, at least in the problem I encountered above:
Following the same approach as PR #74121, I at least have the sample compile by adding:

```
zephyr/samples/drivers/flash_shell/boards/arduino_portenta_h7_stm32h747xx_m4.overlay

/ {
	chosen {
		zephyr,flash-controller = &flash;
	};
};

&flash1 {
	bank2-flash-size = <128>;
};
```

tomi-font commented:
> Do you have an example of a test where the m4 is built and fails due to this?
>
> I see the same problem running the following:
>
> west build -b arduino_portenta_h7/stm32h747xx/m4 -s zephyr/samples/drivers/flash_shell/

Yeah, a test I made on a PR I am working on. I think it fails the same way as the test you mentioned.

> An alternate solution, at least in the problem I encountered above:
Following the same approach as PR https://github.com/zephyrproject-rtos/zephyr/pull/74121, I at least have the sample compile by adding:

Yeah. The thing is, why would this need to be added to every test/sample that doesn't want to exclude this board target? Cannot it be part of the board definition? My goal is to support board targets based on generic filters, without needing to add anything specific such as this.
This board target is the only one (apart from another one for which I could make a proper fix) that slips through [the filtering](https://github.com/zephyrproject-rtos/zephyr/pull/76222/files#diff-fb5682f6af21bb48291e3d8c7e6c7a2ba99a9601dfc2e97f4deabe459424350dR9-R12) (when running `twister --all`) and doesn't compile due to the missing `bank2-flash-size` property that the driver is complaining about.

> I may be wrong here, but I don't think disabling the flash driver on board level is what we want.

I agree it's a bit drastic. I did that because I wasn't comfortable defining `bank2-flash-size` at the board level and wanted to bring this up with relevant people.

There clearly seems to be something wrong with this board because it's excluded in samples/tests that make use of the flash (`samples/drivers/flash_shell`, `tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported`).

Could `bank2-flash-size` be defined at the board level?

benjaminbjornsson commented:
> Cannot it be part of the board definition?

It probably could? These lines seem to be obsolete since m4 flash controller support has been added?
https://github.com/zephyrproject-rtos/zephyr/blob/ca48767be48ca6a7536edf7b1d59f3246046d40e/dts/arm/st/h7/stm32h747Xi_m4.dtsi#L14-L15

and then add the bank2-flash-size property at a relevant ST dtsi level?

> There clearly seems to be something wrong with this board because it's excluded in samples/tests that make use of the flash (`samples/drivers/flash_shell`, `tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported`).

Looking at the exclude list of `tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported` it looks like there's an STM32 H7-series problem.

tomi-font committed:
dts: arm: st: h7: fix flash on M4 board targets

The flash controller is nowadays supported on the M4 core.
Add the bank2-flash-size property to the board definitions as required
by the STM32 H7 flash driver.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

tomi-font commented:
> > Cannot it be part of the board definition?
>
> It probably could?

Yes, seems like it does. I managed to get the sample/test where the STM32 H7 M4 board targets were excluded to compile on them. Thanks for your input!

benjaminbjornsson commented:
Great! Move the `bank2-flash-size` to the flash1 node that is declared in the same dtsi as they are referenced. Otherwise LGTM.

benjaminbjornsson reviewed (approved):
LGTM

erwango reviewed (approved):
Thanks for this fix
